build-system/tasks/typescript-compile.js

```diff
@@ -0,0 +1,45 @@
+/**
```

Comment:
Since this file is related to the server and isn't a top-level `gulp` task, it should live under `build-system/server`.

Reply:
i moved it out since this will most likely be used across the board now (not just server related work), would it still make sense to leave it there?

Reply:
Right now, all JS files under `tasks/` are top-level gulp tasks. Until such time that this becomes a top-level task, I think it should live under `server/`.